tools/android/install_base.sh

```diff
@@ -0,0 +1,295 @@
+    local field_name=${1}
+    # shellcheck source=/etc/os-release
+    (source /etc/os-release &> /dev/null && printf "%s" "${!field_name}")
+}
+
+# Test the value of a field in /etc/os-release
+test_os_release(){
+    local field_name=${1}
+    local value=${2}
+
+    if [[ "$(read_os_release "${field_name}")" == "${value}" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+
+ANDROID_HOME="$(dirname "${0}")/android-sdk-linux"
+export ANDROID_HOME
+export ANDROID_USER_HOME="${ANDROID_HOME}/.android"
+
+mkdir -p "${ANDROID_HOME}/cmdline-tools"
+
+cleanup_android_home() {
+    echo "Cleaning up Android SDK: ${ANDROID_HOME}"
+    rm -rf "${ANDROID_HOME}"
+    mkdir -p "${ANDROID_HOME}/cmdline-tools"
+}
+
+install_android_sdk_manager() {
+    echo "Installing Android SDK manager ..."
+
+    # URL taken from "Command line tools only": https://developer.android.com/studio
+    local url="https://dl.google.com/android/repository/commandlinetools-linux-11076708_latest.zip"
+
+    echo "Downloading Android SDK manager from: $url"
+    wget -qO- "${url}" | bsdtar -xf- -C "${ANDROID_HOME}/cmdline-tools"
+
+    echo "Moving commandlinetools to cmdline-tools/latest..."
+    # First, clear cmdline-tools/latest if it exists.
+    rm -rf "${ANDROID_HOME}/cmdline-tools/latest"
+    mv "${ANDROID_HOME}/cmdline-tools/cmdline-tools" "${ANDROID_HOME}/cmdline-tools/latest"
+
+    chmod +x -R "${ANDROID_HOME}/cmdline-tools/latest/bin"
+
+    yes | (call_android_sdkmanager --licenses || true)
+
+    echo "Creating the link to skins directory..."
+    readlink "${ANDROID_HOME}/skins" > /dev/null 2>&1 || ln -sf "../skins" "${ANDROID_HOME}/skins"
+}
+
+# Android SDK is picky on Java version, so we need to set JAVA_HOME manually.
+# In most distributions, Java is installed under /usr/lib/jvm so use that.
+# according to the distribution
+ANDROID_SDK_JAVA_VERSION=17
+find_java_home() {
+    _JAVA_BIN=$(find -L /usr/lib/jvm -path "*${ANDROID_SDK_JAVA_VERSION}*/bin/java" -not -path '*/jre/bin/*' -print -quit)
+    _JAVA_HOME=$(dirname "${_JAVA_BIN}")/../
+
+    echo "Found JAVA_HOME=${_JAVA_HOME}"
+}
+
+call_android_sdk() {
+    local tool="${ANDROID_HOME}/cmdline-tools/latest/bin/${1}"
+    shift
+    JAVA_HOME=${_JAVA_HOME} "${tool}" "$@"
+}
+
+call_android_sdkmanager() {
+    call_android_sdk sdkmanager "$@"
+}
+
+call_android_avdmanager() {
+    call_android_sdk avdmanager "$@"
+}
+
+# Needs install_android_sdk_manager first
+install_android_tools() {
+    yes | call_android_sdkmanager --verbose --channel=0 --install "platform-tools"
+    yes | call_android_sdkmanager --verbose --channel=0 --install "platforms;android-31"
+    yes | call_android_sdkmanager --verbose --channel=0 --install "platforms;android-34"
+    yes | call_android_sdkmanager --verbose --channel=0 --install "system-images;android-31;google_apis;${HOST_ARCH}"
+    yes | call_android_sdkmanager --verbose --channel=0 --install "system-images;android-34;google_apis;${HOST_ARCH}"
```

Comment:
I use `HOST_ARCH` here, but I believe Android SDK supports only `arm64-v8a` Arm variant. Maybe I should use `arm64-v8a` if the host is not an Intel machine.